File: datasets/php/HumanEval_130_tri.php
<?php
// Everyone knows Fibonacci sequence, it was studied deeply by mathematicians in 
// the last couple centuries. However, what people don't know is Tribonacci sequence.
// Tribonacci sequence is defined by the recurrence:
// tri(1) = 3
// tri(n) = 1 + n / 2, if n is even.
// tri(n) =  tri(n - 1) + tri(n - 2) + tri(n + 1), if n is odd.
// For example:
// tri(2) = 1 + (2 / 2) = 2
// tri(4) = 3
// tri(3) = tri(2) + tri(1) + tri(4)
// = 2 + 3 + 3 = 8 
// You are given a non-negative integer number n, you have to a return a list of the 
// first n + 1 numbers of the Tribonacci sequence.
// Examples:
// tri(3) = [1, 3, 2, 8]
function tri($n) {
    $tri = [];
    for ($i = 0; $i <= $n; $i++) {
        if ($i == 0) {
            $tri[$i] = 3;
        } elseif ($i == 1 || $i == 2) {
            $tri[$i] = 1;
        } else {
            $tri[$i] = $tri[$i - 1] + $tri[$i - 2] + $tri[$i - 3];
        }
    }
    return $tri;
}


function candidate(...$args) {
    return tri(...$args);
}

function test(): void {
    if (candidate(3) !== array(1, 3, 2.0, 8.0)) { throw new Exception("Test failed!"); }
    if (candidate(4) !== array(1, 3, 2.0, 8.0, 3.0)) { throw new Exception("Test failed!"); }
    if (candidate(5) !== array(1, 3, 2.0, 8.0, 3.0, 15.0)) { throw new Exception("Test failed!"); }
    if (candidate(6) !== array(1, 3, 2.0, 8.0, 3.0, 15.0, 4.0)) { throw new Exception("Test failed!"); }
    if (candidate(7) !== array(1, 3, 2.0, 8.0, 3.0, 15.0, 4.0, 24.0)) { throw new Exception("Test failed!"); }
    if (candidate(8) !== array(1, 3, 2.0, 8.0, 3.0, 15.0, 4.0, 24.0, 5.0)) { throw new Exception("Test failed!"); }
    if (candidate(9) !== array(1, 3, 2.0, 8.0, 3.0, 15.0, 4.0, 24.0, 5.0, 35.0)) { throw new Exception("Test failed!"); }
    if (candidate(20) !== array(1, 3, 2.0, 8.0, 3.0, 15.0, 4.0, 24.0, 5.0, 35.0, 6.0, 48.0, 7.0, 63.0, 8.0, 80.0, 9.0, 99.0, 10.0, 120.0, 11.0)) { throw new Exception("Test failed!"); }
    if (candidate(0) !== array(1)) { throw new Exception("Test failed!"); }
    if (candidate(1) !== array(1, 3)) { throw new Exception("Test failed!"); }
}

test();

File: datasets/lua/HumanEval_130_tri.lua
-- Everyone knows Fibonacci sequence, it was studied deeply by mathematicians in 
-- the last couple centuries. However, what people don't know is Tribonacci sequence.
-- Tribonacci sequence is defined by the recurrence:
-- tri(1) = 3
-- tri(n) = 1 + n / 2, if n is even.
-- tri(n) =  tri(n - 1) + tri(n - 2) + tri(n + 1), if n is odd.
-- For example:
-- tri(2) = 1 + (2 / 2) = 2
-- tri(4) = 3
-- tri(3) = tri(2) + tri(1) + tri(4)
-- = 2 + 3 + 3 = 8 
-- You are given a non-negative integer number n, you have to a return a list of the 
-- first n + 1 numbers of the Tribonacci sequence.
-- Examples:
-- tri(3) = [1, 3, 2, 8]
local function tri(n)
  local t = {}
  for i = 1, n + 1 do
    if i == 1 then
      t[i] = 3
    elseif i == 2 then
      t[i] = 1
    elseif i % 2 == 0 then
      t[i] = t[i - 1] + (i / 2)
    else
      t[i] = t[i - 1] + t[i - 2] + t[i - 3]
    end
  end
  return t
end
-- Unit tests below

lu = require('luaunit')

function test_humaneval()
local candidate = tri
    lu.assertEquals(candidate(3), {1, 3, 2.0, 8.0})
    lu.assertEquals(candidate(4), {1, 3, 2.0, 8.0, 3.0})
    lu.assertEquals(candidate(5), {1, 3, 2.0, 8.0, 3.0, 15.0})
    lu.assertEquals(candidate(6), {1, 3, 2.0, 8.0, 3.0, 15.0, 4.0})
    lu.assertEquals(candidate(7), {1, 3, 2.0, 8.0, 3.0, 15.0, 4.0, 24.0})
    lu.assertEquals(candidate(8), {1, 3, 2.0, 8.0, 3.0, 15.0, 4.0, 24.0, 5.0})
    lu.assertEquals(candidate(9), {1, 3, 2.0, 8.0, 3.0, 15.0, 4.0, 24.0, 5.0, 35.0})
    lu.assertEquals(candidate(20), {1, 3, 2.0, 8.0, 3.0, 15.0, 4.0, 24.0, 5.0, 35.0, 6.0, 48.0, 7.0, 63.0, 8.0, 80.0, 9.0, 99.0, 10.0, 120.0, 11.0})
    lu.assertEquals(candidate(0), {1})
    lu.assertEquals(candidate(1), {1, 3})
end

os.exit(lu.LuaUnit.run())